server/app/views/admin/apikeys/ApiKeyNewOneView.java

```diff
@@ -0,0 +1,158 @@
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.p;
+import static j2html.TagCreator.text;
+
+import annotations.BindingAnnotations.EnUsLang;
+import com.github.slugify.Slugify;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import controllers.admin.routes;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import java.util.Optional;
+import play.data.DynamicForm;
+import play.i18n.Messages;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.apikey.ApiKeyService;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.admin.AdminLayout;
+import views.components.FieldWithLabel;
+import views.components.LinkElement;
+import views.style.Styles;
+
+/** Renders a page for adding a new ApiKey. */
+public final class ApiKeyNewOneView extends BaseHtmlView {
+  private final AdminLayout layout;
+  private final Messages enUsMessages;
+  private final Slugify slugifier = new Slugify();
+
+  private static final String EXPIRATION_DESCRIPTION =
+      "Specify a date when this API key will no longer be valid. The expiration date"
+          + " may be edited after the API key is created. It is strongly recommended to"
+          + " allow API keys to expire and create new ones to ensure that if a key"
+          + " has been stolen, malicious users will eventually lose access. For"
+          + " highly sensitive data, expiring keys once a week or once a month is"
+          + " recommended. Once a year is acceptable in most situations.";
+  private static final DomContent[] SUBNET_DESCRIPTION = {
+    text("Specify a subnet using "),
+    new LinkElement()
+        .setText("CIDR notation")
+        .setHref("https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing")
+        .opensInNewTab()
+        .asAnchorText(),
+    text(
+        " that is allowed to call the API. The"
+            + " subnet may be edited after the API key is created. A single IP address"
+            + " can be specified with a mask of /32. For example, \"8.8.8.8/32\""
+            + " allows only the IP 8.8.8.8 to use the API key. All IP addresses can be"
+            + " allowed, but this is strongly discouraged for security reasons because"
+            + " it would allow any computer connected to the internet to use the key"
+            + " if they obtain it. All IP addresses can be allowed with a value of"
+            + " 0.0.0.0/0, though this is not recommended.")
+  };
+
+  @Inject
+  public ApiKeyNewOneView(AdminLayout layout, @EnUsLang Messages enUsMessages) {
+    this.layout = checkNotNull(layout);
+    this.enUsMessages = checkNotNull(enUsMessages);
+  }
+
+  public Content render(Request request, ImmutableSet<String> programNames) {
```

Comment:
Not-blocking: Since there's only one caller of this, consider removing the method overload and having the controller call render(request, programNames, Optional.empty()).

Reply:
There are two callers: the `newOne` method for the initial render, and the `create` method when a submission fails validation -- both `render` method versions get used.